### PR TITLE
initial commit

### PR DESCRIPTION
**Pull request by lleadbet**

### CODE REVIEW

File: sidecar/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+        // Log error in case complexity cannot be calculated (invalid query, misconfiguration, etc.)
+        // @ts-ignore
+        console.error('Could not calculate complexity', e.message);
+        return res.json({
+            "version": 1,
+            "stage": "RouterRequest",
+            "control": {
+                Break: 400
+            },
+            "id": "1b19c05fdafc521016df33148ad63c1b",
```
Comment (lennyburdette) [inline sidecar/src/index.ts]: what's this for?

File: sidecar/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+        if (options.interface && t instanceof GraphQLInterfaceType) {
+            console.log(`Found interface; adding: ${options.interface}`)
+            args.childComplexity += options.interface
+        }
+    }
+}
+const app = express()
+app.use(express.json())
+app.use('*', (req, res) => {
+    if (req.body.body.operationName === 'IntrospectionQuery') {
```
Comment (lennyburdette) [inline sidecar/src/index.ts]: does the complexity code not handle introspection fields? this is an easy bypass

File: sidecar/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+        let hash = crypto.createHash('md5').update(req.body.sdl).digest("hex")
+        if (hash !== schemaHash) {
+            console.log(`New schema, storing in memory...`)
+            schemaHash = hash
+            schema = buildSchema(req.body.sdl)
+            operationMap = new Map()
+        }
+        let complexity = 0;
+        let operationHash: string = crypto.createHash('md5').update(JSON.stringify(req.body.body)).digest("hex")
+        if (!!operationMap.get(operationHash)) {
```
Comment (lennyburdette) [inline sidecar/src/index.ts]: `operationMap.has`?

**Review by lennyburdette — APPROVED** [review]

File: sidecar/src/index.ts (hunk 2 above)
Comment (lleadbet) [reply sidecar/src/index.ts]: I believe it does, but gets a little weird wrt to the type definition evaluation so just short-circuited for now

File: sidecar/src/index.ts (hunk 3 above)
Comment (lleadbet) [reply sidecar/src/index.ts]: ah, good callout. will fix

File: sidecar/src/index.ts (hunk 1 above)
Comment (lleadbet) [reply sidecar/src/index.ts]: The break? Noticed errors happening somewhat and wanted to fail close vs. fail open; just a placeholder really.

Comment (lennyburdette) [reply sidecar/src/index.ts]: sorry no, the id field

Comment (lleadbet) [reply sidecar/src/index.ts]: Ah, didn't realize it was optional from the docs- removed and works just fine.

**Pull request merged**